HopsanCore/src/ComponentSystem.cpp

```diff
@@ -701,66 +701,67 @@ bool ComponentSystem::sortComponentVector(std::vector<Component*> &rComponentVec
-                    {
+                        (pUnsrtComp->getTypeName() == HOPSAN_BUILTIN_TYPENAME_CONDITIONALSUBSYSTEM)) {
                         sortHint = pPort->getInternalSortHint();
                     }
 
-                    if ( (sortHint == Destination) && pPort->isConnected() )
-                    {
-                        Port *pSourcePort = pPort->getNodePtr()->getSortOrderSourcePort();
-                        if (pSourcePort && pSourcePort->getComponent())
-                        {
-                            pRequiredComponent = pSourcePort->getComponent();
+                    if ( (sortHint == Destination) && pPort->isConnected() ) {
+                        std::vector<Port *> sourcePorts;
+                        if(pPort->isMultiPort()) {
+                            for(int s=0; s<pPort->getNumPorts(); ++s) {
+                                sourcePorts.push_back(pPort->getNodePtr(s)->getSortOrderSourcePort());
+                            }
                         }
-                    }
-                    if(pRequiredComponent /*&& (pRequiredComponent->getTypeName() != "SignalUnitDelay")*/)
-                    {
-                        if(pRequiredComponent->mpSystemParent == this)
-                        {
-                            if(!vectorContains<Component*>(newComponentVector, pRequiredComponent) &&
-                                    vectorContains<Component*>(rComponentVector, pRequiredComponent))
-                            {
-                                readyToAdd = false;     //Depending on normal component which has not yet been added
+                        else {
+                            sourcePorts.push_back(pPort->getNodePtr()->getSortOrderSourcePort());
+                        }
+                        for(size_t p=0; p<sourcePorts.size(); ++p) {
+                            if (sourcePorts[p] && sourcePorts[p]->getComponent()) {
+                                pRequiredComponents.push_back(sourcePorts[p]->getComponent());
                             }
                         }
-                        else
-                        {
-                            if(!vectorContains<Component*>(newComponentVector, pRequiredComponent->mpSystemParent) &&
-                                    (pRequiredComponent->mpSystemParent->getTypeCQS() == pPort->getComponent()->getTypeCQS()) &&
-                                    vectorContains<Component*>(rComponentVector,pRequiredComponent->mpSystemParent))
-                            {
-                                readyToAdd = false;     //Depending on subsystem component which has not yet been added
+                    }
+                    if(!pRequiredComponents.empty()) {
+                        for(size_t r=0; r<pRequiredComponents.size(); ++r) {
+                            if(pRequiredComponents[r]->mpSystemParent == this) {
```

Comment:
Not a big deal in this case, but I would prefer adding `Component* pRequiredComponent=pRequiredComponents[r];` and then using pRequiredComponent (like before). Since only the pointer at location r is checked every loop.